### PR TITLE
CPTabView improvements

### PR DESCRIPTION
**Pull request by cacaodev**
NEW: CPTabViewItem -setImage & nib2cib
FIXED: CPTabView -insertTabViewItem:atIndex:
NEW: CPTabView bindings: CPContentBinding, CPSelectedIndexBinding and CPSelectionIndexesBinding with no selection & multiple selection placeholders.
NEW: CPTabView -tabViewItemWithViewController:

With ojtest and manual tests. New test Manual/CPTabViewBindings/

### CODE REVIEW

File: AppKit/CPTabView.j [hunk 1]
```diff
@@ -176,11 +201,10 @@ var CPTabViewDidSelectTabViewItemSelector           = 1 << 1,
...
 - (int)indexOfTabViewItemWithIdentifier:(CPString)anIdentifier
 {
-    for (var index = [_items count]; index >= 0; index--)
-        if ([[_items[index] identifier] isEqual:anIdentifier])
-            return index;
-
-    return CPNotFound;
+    return [[self items] indexOfObjectPassingTest:function(item, idx, stop)
+    {
+        [[item identifier] isEqual:anIdentifier];
```
Comment (Dogild) [inline AppKit/CPTabView.j]: Should be return here

File: AppKit/CPTabView.j [hunk 2]
```diff
@@ -285,22 +309,33 @@ var CPTabViewDidSelectTabViewItemSelector           = 1 << 1,
...
 
+// Like selectTabViewItemAtIndex: but without bindings interaction
+- (BOOL)_selectTabViewItemAtIndex:(CPUInteger)anIndex
+{
     var aTabViewItem = [self tabViewItemAtIndex:anIndex];
 
+    if (aTabViewItem == _selectedTabViewItem)
+        return NO;
+
     if ((_delegateSelectors & CPTabViewShouldSelectTabViewItemSelector) && ![_delegate tabView:self shouldSelectTabViewItem:aTabViewItem])
```
Comment (Dogild) [inline AppKit/CPTabView.j]: We should have a method `_sendDelegateShouldSelecttabViewItem:`

File: AppKit/CPTabView.j [hunk 3]
```diff
@@ -285,22 +309,33 @@ var CPTabViewDidSelectTabViewItemSelector           = 1 << 1,
...
         return NO;
 
     if (_delegateSelectors & CPTabViewWillSelectTabViewItemSelector)
         [_delegate tabView:self willSelectTabViewItem:aTabViewItem];
```
Comment (Dogild) [inline AppKit/CPTabView.j]: Same here, we should have a method `_sendDelegateWillSelecttabViewItem:`

File: AppKit/CPTabView.j [hunk 4]
```diff
@@ -470,30 +509,213 @@ var CPTabViewDidSelectTabViewItemSelector           = 1 << 1,
...
+}
+
+- (void)setItems:(CPArray)tabViewItems
+{
+    if ([tabViewItems isEqualToArray:[_tabs segments]])
+        return;
+
+    [[self items] makeObjectsPerformSelector:@selector(_setTabView:) withObject:nil];
+    [_tabs setSegments:tabViewItems];
+    [tabViewItems makeObjectsPerformSelector:@selector(_setTabView:) withObject:self];
```
Comment (Dogild) [inline AppKit/CPTabView.j]: Do we really need to do that ? Can we optimize that ?

File: AppKit/CPTabViewItem.j [hunk 5]
```diff
@@ -55,12 +56,32 @@ CPPressedTab    = 2;
...
     unsigned    _tabState;      // Looks like it is not yet implemented
+
+    CPImage     _image @accessors(property=image);
+    CPViewController _viewController @accessors(getter=viewController);
+
+    BOOL        _enabled @accessors(property=enabled);
+    BOOL        _selected @accessors(property=selected);
+    CGRect      _tabRect @accessors(property=frame);
+    float       _width @accessors(property=width);
+}
```
Comment (Dogild) [inline AppKit/CPTabViewItem.j]: Style

**Comment by cacaodev** [comment]
There is a layout issue with the border of the last segment after an insertion.
We need to check if it is a CPSegmentedControl issue and if there is a regression here.

**Pull request merged**